api/v1beta1/verticadb_conversion_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+ (c) Copyright [2021-2023] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+//nolint:dupl
+package v1beta1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "github.com/vertica/vertica-kubernetes/api/v1"
+	vmeta "github.com/vertica/vertica-kubernetes/pkg/meta"
+)
+
+var _ = Describe("verticadb_conversion", func() {
+	const trueStrVal = "true"
+	It("should convert ignoreClusterLease", func() {
+		v1beta1VDB := MakeVDB()
+		v1VDB := v1.VerticaDB{}
+		v1beta1VDB.Spec.IgnoreClusterLease = true
+		Ω(v1beta1VDB.ConvertTo(&v1VDB)).Should(Succeed())
+		Ω(v1VDB.Annotations[vmeta.IgnoreClusterLeaseAnnotation]).Should(Equal(trueStrVal))
+		v1beta1VDB.Spec.IgnoreClusterLease = false
+		Ω(v1beta1VDB.ConvertTo(&v1VDB)).Should(Succeed())
+		Ω(v1VDB.Annotations[vmeta.IgnoreClusterLeaseAnnotation]).Should(BeEmpty())
+		v1VDB.Annotations[vmeta.IgnoreClusterLeaseAnnotation] = trueStrVal
+		Ω(v1beta1VDB.ConvertFrom(&v1VDB)).Should(Succeed())
+		Ω(v1beta1VDB.Spec.IgnoreClusterLease).Should(BeTrue())
+	})
+
+	It("should convert ignoreUpgradePath", func() {
+		v1beta1VDB := MakeVDB()
+		v1VDB := v1.VerticaDB{}
+		v1beta1VDB.Spec.IgnoreUpgradePath = true
+		Ω(v1beta1VDB.ConvertTo(&v1VDB)).Should(Succeed())
+		Ω(v1VDB.Annotations[vmeta.IgnoreUpgradePathAnnotation]).Should(Equal(trueStrVal))
+		v1beta1VDB.Spec.IgnoreUpgradePath = false
+		Ω(v1beta1VDB.ConvertTo(&v1VDB)).Should(Succeed())
+		Ω(v1VDB.Annotations[vmeta.IgnoreUpgradePathAnnotation]).Should(BeEmpty())
+		v1VDB.Annotations[vmeta.IgnoreUpgradePathAnnotation] = trueStrVal
+		Ω(v1beta1VDB.ConvertFrom(&v1VDB)).Should(Succeed())
+		Ω(v1beta1VDB.Spec.IgnoreUpgradePath).Should(BeTrue())
+	})
+
+	It("should convert RestartTimeout", func() {
```

Comment:
Are you going to add more testcase?

Reply:
Good point. I think I need some tests for transientSubclusterRouting.